### PR TITLE
Feature/ts support

### PR DESCRIPTION
**Pull request by tnobody**
This PR will introduce Typescript support to Sakuli testcases.

Some core adjustments for Sakuli-Testexecution where necessary to implement this feature:

## Typings

A new package `@sakuli/legacy-types` was introduced. Due to the dynamic nature of Sakuli we could not reuse all typings one-to-one into this package, therefore the typings file is created in an semi-automated process.

## Rollup-Hooks

The rollup-hooks were adapted to transpile files with an `.ts` or `.tsx` extension.

### Changes in rollup format

The previous format of rollup was `iife`. During some Testing it turned out that, this format has disadvantages when packages from node_modules or node itself where imported. The rollup-hooks package hacked it in the way that it reads the _unresolved_ modules from rollup and inject them globally into the context. That hack has its shortcomings in context packages with a name that cannot directly used as an valid variable name - like `@sakuli/cool-package`. Long story short: changing that format to `commonjs` and provide the `require` function in the context solves this problem and also gives the chance to remove the package-import hack.

**This changes might come with some side-effects which i couldn't cover right now. Please Test it carefully**

This change is also necessary to implement https://github.com/sakuli/sakuli/issues/187 properly

## enable-typescript command

Is just a simple command which installs the types and adds an tsconfig.json to the users workspace, lower the barrier to use typescript

### CODE REVIEW

File: packages/sakuli-rollup-hooks/src/rollup-lifecycle-hooks.class.ts [hunk 1]
```diff
@@ -1,32 +1,73 @@
...
+
         const {output} = await bundle.generate({
-            format: 'iife',
+            format: 'commonjs',
             sourcemap: true,
             file: 'bundle-rollup.js',
         });
         const [rollupOutput] = output;
-        this.imports = rollupOutput.imports;
+        this.debug(`Bundled testcase file ${filePath} to: `, rollupOutput.code);
```
Comment (s1hofmann) [inline packages/sakuli-rollup-hooks/src/rollup-lifecycle-hooks.class.ts]: Maybe skip logging the whole resulting bundled code.

**Comment by s1hofmann** [comment]
In order to manage possibly multiple API which bring TypeScript support the `@sakuli/rollup-hooks` package might hold a mapping `package -> typings` to install the appropriate typings for an API.
Keeping TypeScript support inside the rollup-hooks package allows to manage the `tsconfig.js` file at a central place.

**Comment by tnobody** [comment]
@s1hofmann latest develop is merged. You can start reviewing :)

File: packages/sakuli-rollup-hooks/src/ts-support/enable-typescript-command.class.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    })
+}
+
+export const enableTypescriptCommand: CommandModuleProvider = (): CommandModule => {
+    return ({
+        command: 'enable-typescript [project]',
+        describe: `Enables Typescript support for the provided project`,
+        builder(argv: Argv): any {
+            return argv.positional('project', {
+                describe: 'Path to the project, can be absolute or relative to $PWD'
```
Comment (s1hofmann) [inline packages/sakuli-rollup-hooks/src/ts-support/enable-typescript-command.class.ts]: Maybe we should clarify that with `project` the actual npm project where Sakuli is installed is meant, not a particular test suite etc.

File: package.json [hunk 3]
```diff
@@ -42,5 +42,8 @@
     "typedoc:json": "typedoc --options ../../typedoc.js --json ../../docs/{package}/docs.json --out ../../docs/{package} src/",
     "test:watch": "jest --watch --coverage",
     "test:ci": "jest --runInBand --no-cache --coverage --passWithNoTests"
+  },
+  "dependencies": {
+    "@sakuli/legacy": "latest"
   }
```
[the remark this answers is not in the document]
Comment (tnobody) [reply package.json]: No there is no reason, it is removed in the next commit.

File: packages/sakuli-rollup-hooks/src/ts-support/enable-typescript-command.class.ts (hunk 2 above)
Comment (tnobody) [reply packages/sakuli-rollup-hooks/src/ts-support/enable-typescript-command.class.ts]: I think the term `project` is used in the same way as we use it in the docs on sakuli.io.

But we have to be careful for upcoming features like #22 to use the same wording.

**Pull request merged**